baseplate/sidecars/event_publisher.py

```diff
@@ -198,43 +231,38 @@ def publish_events() -> None:
             },
             "key": {"name": config.String, "secret": config.Base64},
             "max_queue_size": config.Optional(config.Integer, MAX_QUEUE_SIZE),
+            "max_element_size": config.Optional(config.Integer, MAX_EVENT_SIZE),
+            "queue_type": config.Optional(config.String, default=QueueType.POSIX.value),
+            "queue_host": config.Optional(config.String, DEFAULT_HOST),
+            "queue_port": config.Optional(config.Integer, DEFAULT_PORT),
         },
     )
 
     metrics_client = metrics_client_from_config(raw_config)
 
-    event_queue = MessageQueue(
-        "/events-" + args.queue_name,
-        max_messages=cfg.max_queue_size,
-        max_message_size=MAX_EVENT_SIZE,
+    queue_name = f"/events-{args.queue_name}"
+    event_queue: MessageQueue = create_queue(
+        cfg.queue_type, queue_name, cfg.max_queue_size, cfg.max_element_size
     )
 
     # pylint: disable=maybe-no-member
     serializer = SERIALIZER_BY_VERSION[cfg.collector.version]()
     batcher = TimeLimitedBatch(serializer, MAX_BATCH_AGE)
     publisher = BatchPublisher(metrics_client, cfg)
 
-    while True:
-        message: Optional[bytes]
-
-        try:
-            message = event_queue.get(timeout=0.2)
-        except TimedOutError:
-            message = None
-
-        try:
-            batcher.add(message)
-            continue
-        except BatchFull:
-            pass
+    if cfg.queue_type == QueueType.IN_MEMORY.value and isinstance(
+        event_queue, InMemoryMessageQueue
+    ):
+        gevent.monkey.patch_all()
```

Comment:
need to do this for the tracing sidecar too

Reply:
oops, forgot to save and commit all my changes 😭 fixed